keps/sig-auth/4872-harden-kubelet-cert-validation/README.md

```diff
@@ -0,0 +1,390 @@
+
+### Graduation Criteria
+
+#### Alpha
+
+* Add feature flag for gating usage, off by default
+* Add flag to disable extra validation
+* Unit and integration tests
+
+#### Beta
+* Address user reviews and iterate if needed
+* Feature flag on by default
+* Validation enabled for all kind e2e test jobs
+
+#### GA
+* Successful adoption by at least one provider
+
+### Upgrade / Downgrade Strategy
+
+The feature is opt-in and it can be disabled at any time by just not setting the `--enable-kubelet-cert-cn-validation` flag.
+
+### Version Skew Strategy
+
+Not applicable.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate
+  - Feature gate name: `KubeletCertCNValidation`
+  - Components depending on the feature gate: kube-apiserver
+- [x] Other
+  - Describe the mechanism: kube-apiserver command-line flag `--enable-kubelet-cert-cn-validation`
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No. But requires restarting the kube-apiserver.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No.
+
+###### Does enabling the feature change any default behavior?
+
+Enabling the feature gate doesn't change any behavior.
+
+Enabling the validation does change the default certificate validation behavior.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled once enabled by not setting the command-line flag.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+You just get back the new behavior with the extra cert validation, no extra considerations needed.
+
+###### Are there any tests for feature enablement/disablement?
+
+We will add integration tests to validate the enablement/disablement flow. Test cases specified in a previous section.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout can fail if the feature flag is not enabled but the command-line flag is set.
+
+Already running workloads won't be impacted but cluster users won't be able to access the control plane if the cluster is single-node.
+
+###### What specific metrics should inform a rollback?
+
+`kube_apiserver_validation_kubelet_cert_cn_total` can help inform a rollback. A non-zero value for the `failure` label will require investigation: if the rejected requests are going to legitimate nodes, the feature should be rolled back until kubelet serving certificates are reissued.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No. There is no data stored for this feature which persists between upgrade / downgrade, or between enable / disable.
+The feature is purely an API server configuration option.
```

Comment:
This answer is fine for alpha, but  for beta you'll need to describe this testing path.